### ISSUE TITLE
1.2.0 Trigger and Sampler Plugins crash the host

### ISSUE THREAD

**Issue by novalix** [sadko4u/lsp-plugins#223]
Hi,
i hope you 're doing well.

I have build the plugins of the devel tree to sneak a bit into it. It is really great. 

However all Trigger and Sampler plugins let the host crash immediatly when i try to instantiate them. I have tried it on ardour and carla with the lv2 and the vst2 flavours. The standalone versions load just fine.

Here are the details:
Host system is Debian stable (bullseye) amd64
Build configure was `make config FEATURES='lv2 vst2 jack' DEBUG=1`
The build runs just fine as does the installation to the default location. 
The attached file contains the backtrace of instatiating lsp-trigger-mono plugin in carla.
[carla-lsp-trigger-mono-lv2-gdb.txt](https://github.com/sadko4u/lsp-plugins/files/8223722/carla-lsp-trigger-mono-lv2-gdb.txt)

Keep up your great work!

**Comment by sadko4u** [comment]
Hello! That's interesting.
I think I need to check them. There can be a silly error I think.
Also I would like to recommend to try to configure and build with these options and verify if the problem still exists:
```
make config DEVEL=1 FEATURES='lv2 vst2 jack' DEBUG=1
```

The option `DEVEL=1` will check out all recent submodule versions.

**Comment by novalix** [comment]
Rebuild the whole thing and indeed the lv2 plugins load just fine in ardour and carla. Yeah.
Alas, the vst2 plugins (again only those with "sampler" or "trigger" in the name) are causing problems.
The plugins load in carla but when i try to open the gui the host crashes instantly.

I have had a bit of struggling to generate a somewhat valid output of the debugger, since it outputs a plethora of lines but doesn 't write the whole thing to the log file for whatever reason. So i copied the last bits from stdout to the file. Hope that works.
[lsp-trigger-mono-vst.txt](https://github.com/sadko4u/lsp-plugins/files/8225974/lsp-trigger-mono-vst.txt)

**Comment by sadko4u** [comment]
Seems that I have missed something in the VST2 wrapper. I'll dig into it soon.

**Comment by sadko4u** [comment]
Fixed the recursive loop in the VST2 wrapper. Please verify.
To fetch latest changes, you need to issue:

```
make -j N fetch
```

yes, fetch supports multithreaded execution.

**Comment by novalix** [comment]
I can confirm that the vst versions of the Trigger plugins an lsp-sampler-{mono,stereo} now are loading the ui without crashing.
The Multisampler family however still does.
Tell me if you need another backtrace.

**Comment by sadko4u** [comment]
Hmmm. I'll inspect the multi-sampler plugin. Is the problem related to VST2 only?

**Comment by novalix** [comment]
Yes, it 's only the vst version that still is misbehaving. The lv2 version works fine since you provided the fix and standalone (jack) did work anyway.

**Comment by sadko4u** [comment]
Seems that I've finally fixed it. Please try it again.

**Comment by novalix** [comment]
I 've been in a mixing session, sorry for the delay:-)
Did a rebuild and tested all of the remaining suspects in both hosts.
No crash, no failure! 
I think this is going to be an exciting release.

**Comment by sadko4u** [comment]
Please don't close the issues until the new release will be issued.

**Comment by novalix** [comment]
Ah sorry, did the same mistake on the other issue.

**Comment by sadko4u** [comment]
Available in 1.2.0.
